src/frontend/next/src/components/Post/Post.tsx

```diff
@@ -1,28 +1,28 @@
-import React, { useRef, useState } from 'react';
+import { useRef, useState } from 'react';
 
 import useSWR from 'swr';
 import 'highlight.js/styles/github.css';
-import { makeStyles } from '@material-ui/core/styles';
+import { makeStyles, Theme } from '@material-ui/core/styles';
 import { Box, Grid, Typography, ListSubheader, createStyles } from '@material-ui/core';
 
 // For style, need to add:
 // import '../styles/telescope-post-content.css';
 
-interface Props {
+type Props = {
   postUrl: string;
-}
+};
 
-interface DataItem {
+type DataItem = {
```

Comment:
No.  The difference here between `Props`, which is done correctly, and this (which isn't) is that `Props` is internal to this code.  Whereas, we'll share the definition of `Feed` and `Post` throughout the frontend.

We need to create a folder at the same level as `components` and include an `index.ts` file, like this: https://github.com/vercel/next.js/tree/canary/examples/with-typescript/interfaces

Inside it, you can `export` two types:

```ts
export type Feed  = {
  link: string
  author: string
}

export type Post {
  feed: Feed
  id: string
  post: string
  title: string
  updated: string
  url: string
  html: string
}
```

Then you can import and use them here.

Reply:
Thank you for the detail feedback